server/sandbox_run.go

```diff
@@ -445,21 +446,21 @@ func (s *Server) RunPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 		}()
 
 		// Pass the created namespace path to the runtime
-		err = g.AddOrReplaceLinuxNamespace("network", sb.NetNsPath())
+		err = g.AddOrReplaceLinuxNamespace(string(runtimespec.NetworkNamespace), sb.NetNsPath())
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	if namespaceOptions.HostPid {
-		err = g.RemoveLinuxNamespace("pid")
+	if securityContext.GetNamespaceOptions().GetHostPid() {
```

Comment:
Just curious why you're not doing namespaceOptions.GetHostPid() here and namespaceOptions.GetHostIpc() below instead of recalling securityContext.GetNameSpaceOptions()?  I may have missed it, but I don't see namespaceOptions changing while you're in this function.

Reply:
I may have missed it, feel free to open a PR clean it up :)